lib/ecco/event_listener.rb

```diff
@@ -3,15 +3,33 @@ class EventListener
     include com.github.shyiko.mysql.binlog.BinaryLogClient::EventListener
     java_import com.github.shyiko.mysql.binlog.event.EventType
 
+    WRITE_EVENTS  = { EventType::WRITE_ROWS => "WRITE_ROWS", EventType::EXT_WRITE_ROWS => "WRITE_ROWS" }
+    UPDATE_EVENTS = { EventType::UPDATE_ROWS => "UPDATE_ROWS", EventType::EXT_UPDATE_ROWS => "UPDATE_ROWS" }
+    DELETE_EVENTS = { EventType::DELETE_ROWS => "DELETE_ROWS" , EventType::EXT_DELETE_ROWS => "DELETE_ROWS" }
+    QUERY_EVENTS  = { EventType::QUERY => "QUERY"}
+    ROTATE_EVENTS = { EventType::ROTATE => "ROTATE"}
+
+    TABLE_EVENT = EventType::TABLE_MAP
+
     attr_writer :callback
 
-    def initialize(client)
+    def initialize(client, events_of_interest)
       @client = client
+      @event_list = events_of_interest.inject(&:merge)
```

Comment:
Think I mentioned it IRL, but there's also `#reduce` which is the same as `#inject`. It's just a matter of taste of which one prefers the most, so if you like `#inject", let's keep it.

Reply:
I had forgotten about this, I agree that `#reduce` fits better in this case. :)